src/main/java/seedu/duke/ui/AppUi.java

```diff
@@ -278,10 +285,17 @@ public void displayWelcomeMessage() {
     }
 
     /**
-     * Runs the application, starting with displaying the welcome message.
+     * Runs the main application logic.
+     * Displays the welcome message, initializes the storage object,
+     * and starts the command entry process.
      */
     public void run() {
         displayWelcomeMessage();
+
+        // create a new storage object and load the financial list from the file
+        this.mainStorage = new Storage();
+        financialList = mainStorage.loadFromFile();
```

Comment:
Perhaps this should be in the main FinancialBuddy class, then passed into the Ui/logic/parser?

Reply:
I change the place to loadFromFile, please check if it's acceptable